elastic/kibana
# [EEM] Add APIs for managing api keys and enabling entity discovery

## Summary

This PR adds endpoints for managing builtin entity discovery:
- `GET /internal/api/entities/managed/enablement`: check if stored API key exists and is valid, then checks whether builtin definitions are installed and running
- `PUT /internal/api/entities/managed/enablement`: creates API key if none exists, then installs and kickoff the builtin definitions
- `DELETE /internal/api/entities/managed/enablement`: stops and uninstalls builtin definitions, then delete API key
- `GET /internal/api/entities/definition`: returns the definitions installed and their corresponding state `{ installed: bool, running: bool }`

The API key is stored in an encrypted saved object, and has only the permissions required to manage the transforms/ingest pipelines, and read the source data.
Built in definitions are stored in code and have hardcoded IDs with a special prefix `builtin_`.

The change also includes an `entityClient` that can be consumed by public plugins and exposes methods that proxy the http endpoints listed previously.

### Testing
- verify entity discovery is not enabled with `GET kbn:/internal/api/entities/managed/enablement`
- setup entity discovery with `PUT kbn:/internal/api/entities/managed/enablement`
- verify builtin definitions are installed and running `GET kbn:/internal/api/entities/definition?builtIn=true`
- verify entity discovery is enabled with `GET kbn:/internal/api/entities/managed/enablement`
- disable entity discovery with `DELETE kbn:/internal/api/entities/managed/enablement`
- verify builtin definitions are not installed anymore `GET kbn:/internal/api/entities/definition?builtIn=true`
- verify entity discovery is disabled with `GET kbn:/internal/api/entities/managed/enablement`
